index.html

```diff
@@ -1177,119 +1177,306 @@ <h2 id="relationship-to-permissions-policy">
           </p>
-          <pre class="lang-json">
-          {
-            "descriptor": {
-              "name": "midi",
-              "sysex": true
-            },
-            "state": "granted"
-          }
-          </pre>
-        </aside>
+          <section>
+            <h5 id="extension-module-permissions-definition">
+              Definition
+            </h5>
+            <p>
+              [=remote end definition=]
+            </p>
+            <pre class="cddl remote-cddl">
+              PermissionsCommand = (
+                permissions.setPermission
+              )
+            </pre>
+          </section>
+          <section>
+            <h5 id="extension-module-permissions-types">
+              Types
+            </h5>
+            <section>
+              <h6 id="extension-module-type-permissions-PermissionDescriptor">
+                The permissions.PermissionDescriptor Type
+              </h6>
+              <pre class="cddl local-cddl">
+                permissions.PermissionDescriptor = {
+                  type: "permissionDescriptor",
+                  name: text,
+                }
+              </pre>
+              <p>
+                The <code>permissions.PermissionDescriptor</code> type represents a {{PermissionDescriptor}}.
+              </p>
+            </section>
+            <section>
+              <h6 id="extension-module-type-permissions-PermissionState">
+                The permissions.PermissionState Type
+              </h6>
+              <pre class="cddl local-cddl">
+                permissions.PermissionState = "granted" / "denied" / "prompt"
+              </pre>
+              <p>
+                The <code>permissions.PermissionState</code> type represents a {{PermissionState}}.
+              </p>
+            </section>
+          </section>
+          <section>
+            <h5 id="extension-module-permissions-commands">
+              Commands
+            </h5>
+            <section>
+              <h6 id="extension-module-command-permissions-setPermission">
+                The permissions.setPermission Command
+              </h6>
+              <p>
+                The <dfn class="export" data-dfn-for="extension modules">Set Permission</dfn>
+                [=extension module=] simulates user modification of a {{PermissionDescriptor}}'s
+                [=permission state=].
+              </p>
+              <dl>
+                <dt>Command Type</dt>
+                <dd>
+                  <pre class="cddl remote-cddl">
+                    permissions.setPermission = (
+                      method: "permissions.setPermission",
+                      params: permissions.SetPermissionParameters
+                    )
+
+                    permissions.SetPermissionParameters = {
+                      descriptor: permissions.PermissionDescriptor,
+                      state: permissions.PermissionState,
+                      ? context: browsingContext.BrowsingContext,
```

Comment:
todo (as discussed): this would be a user context but we could leave it out at first as it would be a new feature compared with the webdriver classic

Reply:
Done. Removed.